core/include/kernel/dt.h

```diff
@@ -93,6 +95,38 @@ int dt_map_dev(const void *fdt, int offs, vaddr_t *base, size_t *size);
  */
 bool dt_have_prop(const void *fdt, int offs, const char *propname);
 
+/*
+ * Get the DT interrupt property of the @node. In the DT an interrupt
+ * is defined with at least 2x32 bits detailling the interrupt number and type.
+ *
+ * @fdt reference to the Device Tree
+ * @node is the node offset to read
+ *
+ * Returns the interrupt number if value >= 0
+ * otherwise DT_INFO_INVALID_INTERRUPT
+ */
+int dt_get_irq(void *fdt, int node);
+
+/*
+ * Modify or add "status" property to "disabled"
+ *
+ * @fdt reference to the Device Tree
+ * @node is the node offset to modify
+ *
+ * Returns 0 on success or -1 on failure
+ */
+int dt_disable_status(void *fdt, int node);
+
+/*
+ * Force secure-status = "okay" and status="disabled" for the target node.
+ *
+ * @fdt reference to the Device Tree
+ * @node is the node offset to modify
+ *
+ * Returns 0 on success or -1 on failure
+ */
+int dt_enable_secure_status(void *fdt, int node);
```

Comment:
suggest to rename `dt_enable_secure_only_status()`.

Reply:
well... `dt_enable_secure_only_status()` is as confusing as `dt_enable_secure_status()`.
Discard my comment.